noir-projects/noir-protocol-circuits/crates/private-kernel-lib/src/private_kernel_circuit_public_inputs_composer.nr

```diff
@@ -70,6 +71,8 @@ impl PrivateKernelCircuitPublicInputsComposer {
         let _call_request = public_inputs.end.private_call_stack.pop();
         public_inputs.end.public_call_stack = array_to_bounded_vec(start.public_call_stack);
 
+        public_inputs.public_teardown_call_request = previous_kernel_public_inputs.public_teardown_call_request;
```

Comment:
Feels like we need the concept of a struct of "data that was mutable on private but is constant on public", where this would fit, right? Just a thought, I'm definitely not pushing for changing it now though.

Reply:
Yes we are definitely using this concept of "privately mutable", but not in a formal capacity. We're also using the concept of "only write once". I haven't seen enough examples to try to formalize it yet, but it is good to have on the radar.